service/forward_service.cc

```diff
@@ -497,6 +515,9 @@ void forward_service::init_messaging_service() {
     ser::forward_request_rpc_verbs::register_forward_request(
         &_messaging,
         [this](query::forward_request req, std::optional<tracing::trace_info> tr_info) -> future<query::forward_result> {
+            if (_shutdown) {
+                throw std::runtime_error("forward_service is shutting down");
+            }
             return dispatch_to_shards(req, tr_info);
```

Comment:
It looks like you put `if (_sutdown)` checks in front of every invocation of `dispatch_to_shards()` and not more than that. Maybe put a single if into this method only?

Reply:
I have adjusted the positioning of shutdown checks. They now precede the calls to functions that may potentially utilize the messaging service, namely `pager->fetch_page()` and `ser::forward_request_rpc_verbs::send_forward_request()`.

This should hopefully fix the issue of `forward_service` hanging on rpc calls after messaging service shutdown.